src/Compilers/Core/Portable/PEWriter/MetadataVisitor.cs

```diff
@@ -520,9 +532,11 @@ public virtual void Visit(ITypeDefinitionMember typeMember)
             }
             else
             {
-                this.Visit(typeMember.GetAttributes(Context));
-
-                typeMember.Dispatch(this);
+                if (!Context.Filter(typeMember))
```

Comment:
>if (!Context.Filter(typeMember)) [](start = 16, length = 32)

It feels like we shouldn't be changing implementation of virtual methods, someone can override them. I think the strategy should be to ensure the methods are not called for excluded things.